docs/tools/lsp-factoryjs/classes/01-lsp3-universal-profile.md

```diff
@@ -14,42 +14,63 @@ lspFactory.LSP3UniversalProfile.deploy(
-- set the Key Manager as the owner of the LSP0 ERC725 Account.
-- give all [LSP6 permissions](../../../standards/universal-profile/lsp6-key-manager#-types-of-permissions) to the `controllerAddresses` except `DELEGATECALL`.
+- upload metadata to IPFS and set the [LSP3 Universal Profile](../../../standards/universal-profile/lsp3-universal-profile-metadata) metadata,
+- attach the Universal Receiver Delegate to the ERC725 Account contract,
+- set the Key Manager as the owner of the LSP0 ERC725 Account, and
+- set all [LSP6 permissions](../../../standards/universal-profile/lsp6-key-manager#-types-of-permissions) to the `controllerAddresses` except `DELEGATECALL`.
 
-By default the [LSP1 Universal Receiver Delegate](../../../standards/universal-profile/lsp1-universal-receiver-delegate) contract specified in the [versions file](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json) will be attached to the Universal Profile. A custom Universal Receiver Delegate can be optionally deployed by passing custom bytecode inside the ContractDeploymentOptions object. [Read more](../deployment/universal-profile#configuration).
+By default the [LSP1 Universal Receiver Delegate](../../../standards/universal-profile/lsp1-universal-receiver-delegate) contract that is specified in the [versions file](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json) will be attached to the Universal Profile. A custom Universal Receiver Delegate can be optionally deployed, by passing custom bytecode to the ContractDeploymentOptions object. You can read more about it in the[Universal Profile Configuration](../deployment/universal-profile#configuration).
 
 #### Parameters
 
-1. `profileDeploymentOptions` - `Object`: The options used for profile deployment.
-   - `controllerAddresses` - `string[]`: A list of accounts (public addresses) which will be granted [all permissions](../../../../../standards/universal-profile/lsp6-key-manager#-address-permissions) on the newly created Universal Profile.
-   - `lsp3Profile?` - `Object`: If set, the created Universal Profile will be populated with these values.
-     - `name` - `string`: The name of the Universal Profile.
-     - `description` - `string`: The description of the Universal Profile.
-     - `profileImage?` - `File | ImageBuffer | ImageMetadata[]`
-     - `backgroundImage?` - `File | ImageBuffer | ImageMetadata[]`
-     - `tags?` - `string[]`
-     - `links?` - `{title: string, url: string}[]`
-2. `contractDeploymentOptions?` - `Object`: Specify contract deployment details. See [configuration specification](../deployment/universal-profile#configuration) for more information.
+| Name                         | Type   | Description                              |
+| :--------------------------- | :----- | :--------------------------------------- |
+| `contractDeploymentOptions?` | Object | Specifies contract deployment details.   |
```

Comment:
These should be swapped, `profileDeploymentOptions` is the first parameter in the function call and `contractDeploymentOptions` is an optional second parameter

Reply:
I see you have added the optional parameter first in a few places, maybe there is a reason for this?

Reply:
I see you have added the optional parameter first in a few places, maybe there is a reason for this?